### DISCUSSION TITLE
Is support for git submodules planned / on the agenda?

### DISCUSSION THREAD

**Discussion by johanneskastl** [jj-vcs/jj#5205] [Q&A] [answered]
Dear all,

I noticed that [the git compatibility documentation](https://github.com/jj-vcs/jj/blob/main/docs/git-compatibility.md) states that there is no support for git submodules. Unfortunately, it does not link to an open issue to track planned or ongoing development for this feature.

Does anyone know if this support is planned? Or has this been discussed and discarded?

For me this seems to only major blocker from switching my workflow over to `jj`, as I rely heavily on git submodules.

Kind Regards,
Johannes

**Comment by PhilipMetzger** [comment]
Yes, support for it is planned as it is needed for Git compatibility. The work was started about two years ago by then Google Git contributor @glencbz (#1611), but it is now orphaned as Glen has moved on from Google and Source Control work. 

I expect it to be a large block of work with multiple design docs as they require some _specific_ behaviors for the working-copy which needs to be discussed first.

**Reply by johanneskastl** [reply]
Thanks for the information. Is there a issue or feature request tracking the progress?

**Reply by PhilipMetzger** [reply, marked as answer]
look at #1402, #1403, #1404 and #1405

**Reply by johanneskastl** [reply]
Thanks Philip!
